src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/DebuggerBase.java

```diff
@@ -394,7 +396,11 @@ protected long readCompOopAddressValue(long address)
 
   protected long readCompKlassAddressValue(long address)
     throws UnmappedAddressException, UnalignedAddressException {
-    long value = readCInteger(address, getKlassPtrSize(), true);
+    // On 64 bit systems, the compressed Klass* is currently read from the mark
+    // word. We need to load the whole mark, and shift the upper parts.
+    long value = readCInteger(address, getJLongSize(), true);
```

Comment:
`machDesc.getAddressSize()` might be a better fit here instead of `getJLongSize()`?

Reply:
Right. Both is not beautiful, but address size seems less ugly. At some point in the future all of this will go away, anyway.

Reply:
`getAddressSize` does not over-read accidentally on 32-bit :)